streams/readable-streams/async-iterator.any.js

```diff
@@ -625,3 +625,26 @@ for (const preventCancel of [false, true]) {
     rs.getReader();
   }, `return() should unlock the stream synchronously when preventCancel = ${preventCancel}`);
 }
+
+promise_test(async () => {
+  const rs = new ReadableStream({
+    async start(c) {
+      c.enqueue('a');
+      c.enqueue('b');
+      c.enqueue('c');
+      await flushAsyncEvents();
+      // At this point, the async iterator has a read request in the stream's queue for its pending next() promise.
+      // Closing the stream now causes two things to happen *synchronously*:
+      //  1. ReadableStreamClose resolves reader.[[closedPromise]] with undefined.
+      //  2. ReadableStreamClose calls the read request's close steps, which calls ReadableStreamReaderGenericRelease,
+      //     which replaces reader.[[closedPromise]] with a rejected promise.
```

Comment:
Could you check that reader.closed has changed to a different promise?

Reply:
No, I can't check it here. This reader is internal to the async iterator, and the iterator implementation does not use `[[closedPromise]]` in any way.

That said, it doesn't look like this is covered by any of our existing tests for "regular" readers. If you want, I can copy [these tests](https://github.com/web-platform-tests/wpt/blob/9378cebe0d791f016d1b0ecca105e10941d7a34f/streams/readable-streams/default-reader.any.js#L117-L155) and check that `reader.closed` changes after calling `reader.releaseLock()`?

Reply:
> No, I can't check it here. This reader is internal to the async iterator, and the iterator implementation does not use `[[closedPromise]]` in any way.

Oops, yes, sorry.

> That said, it doesn't look like this is covered by any of our existing tests for "regular" readers. If you want, I can copy these tests and check that `reader.closed` changes after calling `reader.releaseLock()`?

That would be great, thanks.